src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -268,8 +415,42 @@ func isInboxWSLInstalled(ctx context.Context, log *logrus.Entry) (bool, bool, er
 		}
 	}
 
-	err = errors.Join(allErrors...)
-	return coreInstalled, kernelInstalled, err
+	return coreInstalled, kernelVersion, errors.Join(allErrors...)
+}
+
+// Get the version of an installed MSI package, given its product code.
+func getMSIVersion(productCode []uint16) (*PackageVersion, error) {
+	version := PackageVersion{}
+	versionStringWide, err := windows.UTF16PtrFromString(INSTALLPROPERTY_VERSIONSTRING)
+	if err != nil {
+		return nil, err
+	}
+	bufSize := uint32(64)
+	wideBuf := make([]uint16, bufSize)
+	for {
+		rv, _, _ := msiGetProductInfo.Call(
+			uintptr(unsafe.Pointer(unsafe.SliceData(productCode))),
+			uintptr(unsafe.Pointer(versionStringWide)),
+			uintptr(unsafe.Pointer(unsafe.SliceData(wideBuf))),
+			uintptr(unsafe.Pointer(&bufSize)),
+		)
+		switch rv {
+		case uintptr(windows.ERROR_SUCCESS):
+			versionString := windows.UTF16ToString(wideBuf[:bufSize])
+			if err = version.UnmarshalText([]byte(versionString)); err != nil {
+				return nil, err
+			}
+			return &version, nil
+		case uintptr(windows.ERROR_MORE_DATA):
+			bufSize = uint32(len(wideBuf) * 2)
+			wideBuf = make([]uint16, bufSize)
```

Comment:
I believe `bufSize` will be set to the required buffer size when the function returns `ERROR_MORE_DATA`; you don't have to keep on guessing what buffer size you need.

I would actually call the function twice: once with size 0 just to query the required size, and the once with the allocated buffer. Get's rid of the loop altogether. If it still fails on 2nd call, it is a fatal error.